releases/release-5.1.4.md

```diff
@@ -0,0 +1,165 @@
+        (dup) - Fix the issue that TiCDC replication task might terminate when the upstream TiDB instance unexpectedly exits [#3061](https://github.com/pingcap/tiflow/issues/3061)
+        (dup) - Fix the issue that TiCDC process might panic when TiKV sends duplicate requests to the same Region [#2386](https://github.com/pingcap/tiflow/issues/2386)
+        (dup) - Fix the issue that the volume of Kafka messages generated by TiCDC is not constrained by `max-message-size` [#2962](https://github.com/pingcap/tiflow/issues/2962)
+        - Nond [#2983](https://github.com/pingcap/tiflow/issues/2983)
+        - Add metrics to observe incremental scan remaining time [#2985](https://github.com/pingcap/tiflow/issues/2985)
+        (dup) - Fix the issue that TiCDC sync task might pause when an error occurs during writing a Kafka message [#2978](https://github.com/pingcap/tiflow/issues/2978)
+
+## Improvements
+
++ TiDB
+
+    - planner: support column range partition pruning for builtin function IN [#26739](https://github.com/pingcap/tidb/issues/26739)
+    - Track the memory usage of IndexJoin more accurate. [#28650](https://github.com/pingcap/tidb/issues/28650)
+
++ TiFlash
+
+    - support functions of ADDDATE() and DATE_ADD() pushed down to tiflash
+
+## Bug Fixes
+
++ TiDB
+
+    - Fix the bug that indexHashJoin may return the error `send on closed channel`. [#31129](https://github.com/pingcap/tidb/issues/31129)
+    (dup) - Fix the data inconsistency issue caused by incorrect usage of lazy existence check and untouched key optimization [#30410](https://github.com/pingcap/tidb/issues/30410)
+    (dup) - Fix the issue that window functions might return different results when using a transaction or not [#29947](https://github.com/pingcap/tidb/issues/29947)
+    (dup) - Fix the issue that the length information is wrong when casting `Decimal` to `String` [#29417](https://github.com/pingcap/tidb/issues/29417)
+    (dup) - Fix the issue that the `GREATEST` function returns inconsistent results due to different values of `tidb_enable_vectorized_expression` (set to `on` or `off`) [#29434](https://github.com/pingcap/tidb/issues/29434)
+    (dup) - Fix the issue that the planner might cache invalid plans for `join` in some cases [#28087](https://github.com/pingcap/tidb/issues/28087)
+
++ TiFlash
+
+    - Fix str_to_date() function incorrectly handles leading zeros when parsing Microseconds
+    - Fix the problem of TiFlash crashing when the memory limit is enabled
+    - Align unix_timestamp behavior with TiDB and mysql when input is earlier than 1970-01-01 00:00:01 UTC
+    - Fix potential data inconsistency when widen pk column type if pk is handle
+    - fix the issue that comparison between Decimal may cause overflow and report `Can't compare`
+    - Fix the issue of unexpected error that `3rd arguments of function substringUTF8 must be constants.`
+    - Fix the issue that TiFlash fails to start up under platform without library `nsl`
+    - release-note
```

Comment:
```suggestion
    - Fix a cast to decimal overflow bug.
    - Fix the bug that castStringAsReal has different behaivor between TiFlash and TiKV/TiDB.
    - Fix random `EstablishMPPConnection` fail after TiFlash server restart.
    - Fix the problem that obsolete data cannot be reclaimed after set TiFlash replica to 0.
    - Fix the inconsistent behavior of `CastStringAsDecimal` between TiFlash and TiKV/TiDB.
    - Fix the bug that results of `where <string>` is wrong because it will be converted to `int` type.
    - Fix TiFlash randomly crash when an MPP query is killed.
    - Fix unexpected error of `Unexpected type of column: Nullable(Nothing)`.
```